spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/AbstractMqttMessageHandler.java

```diff
@@ -109,8 +198,49 @@ protected void incrementClientInstance() {
 	@Override
 	protected void onInit() throws Exception {
 		super.onInit();
+		if (this.topicProcessor instanceof BeanFactoryAware && getBeanFactory() != null) {
+			((BeanFactoryAware) this.topicProcessor).setBeanFactory(getBeanFactory());
+		}
+		if (this.qosProcessor instanceof BeanFactoryAware && getBeanFactory() != null) {
+			((BeanFactoryAware) this.qosProcessor).setBeanFactory(getBeanFactory());
+		}
+		if (this.retainedProcessor instanceof BeanFactoryAware && getBeanFactory() != null) {
+			((BeanFactoryAware) this.retainedProcessor).setBeanFactory(getBeanFactory());
+		}
 		if (this.converter == null) {
-			this.converter = new DefaultPahoMessageConverter(this.defaultQos, this.defaultRetained);
+			DefaultPahoMessageConverter mConverter = new DefaultPahoMessageConverter(
```

Comment:
Ah! Sorry. Didn't notice that `this.converter` isn't of `DefaultPahoMessageConverter` type.
Yep! Never mind then :smile: 
Maybe just name of the variable would be more informative ? :wink: 

Typically we avoid reduction since we are Open Source and our code should be as specific as possible.
Much easier to read it in the future. 

Right, first of all I was confused about variable name.